clickTranslate.js

```diff
@@ -6,13 +6,46 @@ document.addEventListener('click', async function(event) {
     }
 
     if (word) {
+        word = trimPunctuation(word);
         const translation = await translate(word);
         console.log('Clicked word:', word);
         console.log('Translation:', await translate(word));
         saveTranslation(word, translation);
     }
 });
 
+/*
+Function trimPunctuation replaces unnecessary tokens in the target string with
+whitespaces, then trims the remaining whitespaces.
+ */
+function trimPunctuation(word) {
+    let finalString = "";
+
+    for (let i = 0; i < word.length; i++) {
+        const tokenFound = tokensInString(word.charAt(i));
```

Comment:
the tokens in string function seems redundant to me, you can probably just use `str.replace` with the existing regex

https://stackoverflow.com/questions/6555182/remove-all-special-characters-except-space-from-a-string-using-javascript

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace